lib/tcx.js

```diff
@@ -0,0 +1,207 @@
+import { nodeVal, get1 } from "./shared";
+
+const attributeNames = [
+  // Trackpoint attributes
+  ["heartRate", "heartRates"],
+  ["cadence", "cadences"],
+  // Extended Trackpoint attributes
+  ["Speed", "speeds"],
+  ["Watts", "watts"],
+  // Lap attributes
+  ["TotalTimeSeconds", "totalTimeSeconds"],
+  ["DistanceMeters", "distanceMeters"],
+  ["MaximumSpeed", "maxSpeed"],
+  ["AverageHeartRateBpm", "avgHeartRate"],
+  ["MaximumHeartRateBpm", "maxHeartRate"],
+  // Extended Lap attributes
+  ["AvgSpeed", "avgSpeed"],
+  ["AvgWatts", "avgWatts"],
+  ["MaxWatts", "maxWatts"],
+];
+
+function getProperties(node) {
+  const prop = {};
+  const extensions = node.getElementsByTagNameNS(
+    "http://www.garmin.com/xmlschemas/ActivityExtension/v2",
+    "*"
+  );
+  attributeNames.forEach((attr) => {
+    const raw = get1(node, attr[0]);
+    if (raw !== null) {
+      prop[attr[1]] = parseFloat(nodeVal(raw));
+    } else if (extensions !== null) {
+      for (let i = 0; i < extensions.length; i++) {
```

Comment:
This is looking a bit problematic: we're looping through `attributeNames`, then a nested loop through each element of `extensions` if the attribute is not found. In the worst case, where there are a lot of extensions, it's a bunch of tight loops of wasted work.

I'm not sure which case this is trying to handle - that `get1` might fail to get an element but getElementsByTagNameNS will succeed. Possibly we should _always_ use one or the other, or separate these loops?